pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -0,0 +1,465 @@
+	if activeGW != nil && activeGW.Name == c.installedActiveGWName {
+		if err := c.syncInterClusterFlows(activeGW); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	if c.installedActiveGWName != "" {
+		if err := c.cleanupFlows(c.installedActiveGWName); err != nil {
+			return err
+		}
+	}
+
+	if activeGW != nil {
+		if activeGW.Name == c.nodeConfig.Name {
+			err = c.installFlowsOnGateway(activeGW)
+		} else {
+			err = c.installFlowsOnRegularNode(activeGW)
+		}
+		if err != nil {
+			return err
+		}
+		c.installedActiveGWName = activeGW.Name
+	} else {
+		c.installedActiveGWName = ""
+	}
+	return nil
+}
+
+func (c *MCRouteController) syncInterClusterFlows(activeGW *mcv1alpha1.Gateway) error {
+	var installedCIImports []*mcv1alpha1.ClusterInfoImport
+	installedCIImportObjects := c.installedCIImports.List()
+	for _, ciImp := range installedCIImportObjects {
+		installedCIImports = append(installedCIImports, ciImp.(*mcv1alpha1.ClusterInfoImport))
+	}
+	desiredCIImports, err := c.ciImportLister.ClusterInfoImports(c.namespace).List(labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	for _, ciImport := range desiredCIImports {
+		if err := c.addMCFlowsForCIImps(activeGW, ciImport); err != nil {
+			return err
+		}
+		c.installedCIImports.Add(ciImport)
+	}
+
+	for _, ciImport := range installedCIImports {
+		if !c.containsCIImport(desiredCIImports, ciImport) {
+			if err := c.deleteMCNodeFlowsForCIImport(ciImport, activeGW.Name); err != nil {
+				return err
+			}
+			c.installedCIImports.Delete(ciImport)
+		}
+	}
+	return nil
+}
+
+func (c *MCRouteController) containsCIImport(ciImports []*mcv1alpha1.ClusterInfoImport, ciImport *mcv1alpha1.ClusterInfoImport) bool {
+	for _, cii := range ciImports {
+		if cii.Name == ciImport.Name {
+			return true
+		}
+	}
+	return false
+}
+
+func (c *MCRouteController) cleanupFlows(gatewayName string) error {
+	if err := c.deleteMCNodeFlowsForGateway(gatewayName); err != nil {
+		return err
+	}
+	if err := c.ofClient.UninstallMulticlusterFlows(gatewayName); err != nil {
+		return err
+	}
+	klog.V(2).InfoS("Deleted flows for Gateway", "gateway", gatewayName)
+	return nil
+}
+
+func (c *MCRouteController) installFlowsOnGateway(gw *mcv1alpha1.Gateway) error {
+	// Install Gateway flows only when the Node is the active Gateway.
+	if err := c.ofClient.InstallMulticlusterClassifierFlows(gw.Name, config.DefaultTunOFPort, true); err != nil {
+		return err
+	}
+	return c.addMCFlowsForCIImps(gw, nil)
+}
+
+func (c *MCRouteController) installFlowsOnRegularNode(gw *mcv1alpha1.Gateway) error {
+	if err := c.ofClient.InstallMulticlusterClassifierFlows(gw.Name, config.DefaultTunOFPort, false); err != nil {
+		return err
+	}
+	return c.addMCFlowsForCIImps(gw, nil)
+}
+
+func (c *MCRouteController) addMCFlowsForCIImps(activeGW *mcv1alpha1.Gateway, ciImp *mcv1alpha1.ClusterInfoImport) error {
+	var err error
+	var allciImports []*mcv1alpha1.ClusterInfoImport
+	if ciImp != nil {
```

Comment:
Do you think easier to understand if we define func for line 366 to 402 that takes a single CIImport, which can be called by both addMCFlowsForCIImps() and the single CIImport case, compared to sharing addMCFlowsForCIImps() for both single CIImport and all CIImports?